### ISSUE TITLE
BFS and DFS in Neo4j?

### ISSUE THREAD

**Issue by HADDADENICAR** [neo4j-contrib/neo4j-graph-algorithms#727]
I want to test BFS and DFS algorithms at my Neo4j graph. [In this article](https://neo4j.com/blog/graph-algorithms-neo4j-15-different-graph-algorithms-and-what-they-do/) i find A listing of Algorithms including BFS and DFS, but no links to a plugin Github repository where i can find documentation about them ! ?
have you implemented this two algorithms in your project ?

**Comment by mneedham** [comment]
Hi @HADDADENICAR,

We do have versions of them but they aren't exposed as procedures themselves - rather they are being used by other algorithms. 

Do you have a use case where you'd need them directly? If you can explain it in a bit more detail it'll help us prioritise what we work on next.

Cheers, Mark

**Comment by HADDADENICAR** [comment]
Yes , actually i m working in Taxi trajectories data, i have created a geohash graph of my data.
Given a starting **geohash** vertex, the outcome tree obtained after running DFS &/or BFS illustrates all likely trips from that place .

**Comment by MohamedElbanna** [comment]
Any updates on the implementation of BFS directly as I also do need it?

**Comment by mneedham** [comment]
Hi,

We've added that to the top of the list of our prioritised things to work on next. Will let you know when we get to it.

Cheers, Mark

**Comment by mneedham** [comment]
Hi [HADDADENICAR](/HADDADENICAR), 

Belatedly we are working on BFS and DFS procedures now. We need to come up with termination conditions for these algorithms. This is what we have so far:

Total length of path

(Show me where I can get to from London going via 3 other cities)

```
MATCH (startNode:Location {name: "London"})
CALL algo.bfs('Location', 'CONNECTED_TO', startNode,
  { totalLength: 3 })
YIELD nodeIds
RETURN nodeIds
```

Total sum of weights

(Show me where I can get to from London within a maximum distance of 45km)

```
MATCH (startNode:Location {name: "London"})
CALL algo.bfs('Location', 'CONNECTED_TO', startNode,
  { totalSumOfWeights: 45 })
YIELD nodeIds
RETURN nodeIds
```

Set of end nodes

(Show me the path from London to Manchester and Dresden)

```
MATCH (startNode:Location {name: "London"})
MATCH (endNode:Location) WHERE endNode.name IN ["Manchester", "Dresden"]
WITH startNode, collect(endNode) AS endNodes

CALL algo.bfs('Location', 'CONNECTED_TO', startNode,
  { endNodes: endNodes })
YIELD nodeIds
RETURN nodeIds
```

Would that work for solving your use case, or if not could you suggest what we'd need to do to make sure you can use these procedures once they're implemented?

**Comment by HADDADENICAR** [comment]
That's so Great ! yes it solves my use case. 
is already under test ? from which version i can use it ?
thank you

**Comment by mneedham** [comment]
@HADDADENICAR it's on this PR, but not finished yet - https://github.com/neo4j-contrib/neo4j-graph-algorithms/pull/826
